closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -38,10 +39,14 @@
 #include "jdk_crypto_jniprovider_NativeCrypto.h"
 #include "NativeCrypto_md.h"
 
-#define OPENSSL_VERSION_1_0 "OpenSSL 1.0."
-#define OPENSSL_VERSION_1_1 "OpenSSL 1.1."
+#define OPENSSL_VERSION_CODE(major, minor) (((major) << 28) | ((minor) << 20))
```

Comment:
I suggest this macro should accept four arguments so it can be used to avoid code duplication (the parameter names are taken from [opensslv.h](https://github.com/openssl/openssl/blob/OpenSSL_1_1_1t/include/openssl/opensslv.h#L19)):
```c
#define OPENSSL_VERSION_CODE(major, minor, fix, patch) \
	(((major) << 28) | ((minor) << 20) | ((fix) << 12) | (patch))
```
The updated macro can then be used in `extractVersionToJlong()` with the locals renamed to match:
```c
    return (jlong)OPENSSL_VERSION_CODE(major, minor, fix, patch);
```

Reply:
Edited.

Reply:
If we changed to this format, it seems that the version which only contains major and minor will return odd values. Check the following results.
```
1.0 formatted: FFFFFFFF
1.0.0 formatted: 10000000
1.1.1a formatted: 10101001
1.1.1k formatted: 1010100B
1.1.1.z formatted: 10101000
2.0 formatted: FFFFFFFF
5.0 formatted: FFFFFFFF
10.0 formatted: FFFFFFFF
10.1 formatted: FFFFFFFF
```